packages/server/lib/routes.js

```diff
@@ -0,0 +1,96 @@
+const path = require('path')
+const la = require('lazy-ass')
+const check = require('check-more-types')
+const debug = require('debug')('cypress:server:routes')
+
+const CacheBuster = require('./util/cache_buster')
+const spec = require('./controllers/spec')
+const reporter = require('./controllers/reporter')
+const runner = require('./controllers/runner')
+const xhrs = require('./controllers/xhrs')
+const client = require('./controllers/client')
+const files = require('./controllers/files')
+const staticCtrl = require('./controllers/static')
+
+module.exports = function (app, config, request, getRemoteState, getDeferredResponse, project, networkProxy) {
+  // routing for the actual specs which are processed automatically
+  // this could be just a regular .js file or a .coffee file
+  app.get('/__cypress/tests', (req, res, next) => {
+    // slice out the cache buster
+    const test = CacheBuster.strip(req.query.p)
+
+    return spec.handle(test, req, res, config, next, project)
```

Comment:
not 100% sure that this controller/other controllers need to be `return`ed, do we use some express middleware that makes returning a promise from a controller meaningful?

Reply:
Probably not. I'll remove them. Any issues should be caught be tests.